pkg/operator/injector/injector.go

```diff
@@ -0,0 +1,487 @@
+	// check whether the configmap is existed
+	err := kubeclient.Get(ctx, client.ObjectKey{Namespace: namespace, Name: configmapName}, configmap)
+	// if configmap not exist , get configmap from defaultConfigmapNamespace
+	if err != nil {
+		defaultConfigmap := &corev1.ConfigMap{}
+		if err := kubeclient.Get(ctx, client.ObjectKey{Namespace: defaultConfigmapNamespace,
+			Name: AnnoConfigmapName.DefaultValue}, defaultConfigmap); err != nil {
+			log.Info(err.Error())
+			s.injectErrorAnnotation(annotation, fmt.Sprintf("get configmap %s from namespace %s error[%s]",
+				AnnoConfigmapName.DefaultValue, defaultConfigmapNamespace, err.Error()))
+			return false
+		}
+		// create new configmap and update namespace
+		injectConfigmap := corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      configmapName,
+				Namespace: namespace,
+			},
+			Data: defaultConfigmap.Data,
+		}
+
+		if err := kubeclient.Create(ctx, &injectConfigmap); err != nil {
+			log.Info(err.Error())
+			s.injectErrorAnnotation(annotation, fmt.Sprintf("create configmap %s in namespace %s error[%s]",
+				configmapName, namespace, err.Error()))
+			return false
+		}
+	}
+	return true
+}
+
+// PodInjector injects agent into Pods
+type PodInjector struct {
+	Client  client.Client
+	decoder *admission.Decoder
+}
+
+// Handle will process every coming pod under the
+// specified namespace which labeled "swck-injection=enabled"
+func (r *PodInjector) Handle(ctx context.Context, req admission.Request) admission.Response {
+	pod := &corev1.Pod{}
+
+	err := r.decoder.Decode(req, pod)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	// init Annotations to store the overlaied value
+	as := NewAnnotations()
+
+	// init SidecarInjectField and get injected strategy from annotations
+	s := NewSidecarInjectField()
+	s.GetInjectStrategy(pod.Spec.Containers, &pod.ObjectMeta.Labels, &pod.ObjectMeta.Annotations)
+
+	if s.needInject {
+		log.Info("will inject agent,please wait for a moment!")
+		// the overlation function is always on
+		s.OverlaySidecar(as, &pod.ObjectMeta.Annotations)
+		switch {
+		// only if user choose to overlay agent config in annotations, then start agent and plugins' overlay
+		case s.agentOverlay && (!s.OverlayAgentConfig(as, &pod.ObjectMeta.Annotations) ||
+			!s.OverlayPluginsConfig(&pod.ObjectMeta.Annotations)):
+			log.Info("overlay agent config error!please look the error annotation!")
+		// configmap will always overwrite the original agent.config in image
+		case !s.CreateConfigmap(ctx, r.Client, req.Namespace, &pod.ObjectMeta.Annotations):
+			log.Info("create configmap error!please look the error annotation!")
+		// add jvmStr to EnvVar and do real injection
+		default:
+			if s.jvmAgentConfigStr != "" {
+				s.env.Value = strings.Join([]string{s.env.Value, s.jvmAgentConfigStr}, "=")
+			}
+			s.Inject(pod)
+			log.Info("inject successfully!")
+		}
```

Comment:
This piece is hard to understand, I have to read your comment to know the logic, that means ppl can't maintain it.

I suggest using `if-else` or https://en.wikipedia.org/wiki/Chain-of-responsibility_pattern to refactor them.